### PR TITLE
Add TypeScript Definitions

### PR DESCRIPTION
**Pull request by ostapenko-me**
Fixed issue #174

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+		protocol?: string;
+		host: string;
+		port?: string;
+		username?: string;
+		password?: string;
+		apiVersion?: string;
+		base?: string;
+		intermediatePath?: string;
+		strictSSL?: boolean;
+		request: Function;
```
Comment (edude03) [inline index.d.ts]: should be `request?: Function;`

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,293 @@
+/// <reference types="request" />
+/// <reference types="node" />
+
+import * as request from 'request';
```
Comment (edude03) [inline index.d.ts]: Not going to argue this one, but it extends `request-promise` which I don't think has types anyway

Comment (KOPTE3) [inline index.d.ts]: From `request` I use only `request.CoreOptions` and `request.RequestResponse`: [here](https://github.com/jira-node/node-jira-client/pull/194/files#diff-b52768974e6bc0faccb7d4b75b162c99R180)

And these types are the same for `request` and `request-promise` packages

**Comment by crhistianr** [comment]
Any chance we can get these merged in? Would be nice to have

**Comment by zlepper** [comment]
While we are waiting for this pull request to be accepted, is it possible to get these types published to the definitely types repos?

**Comment by KOPTE3** [comment]
@zlepper, that's a good idea. I'll do it this week.

**Comment by delete** [comment]
Any update on this, @KOPTE3 ?

**Comment by orta** [comment]
I've created a PR on definitely typed for this 👍

**Pull request closed without merging**